services/kinesis/src/test/java/software/amazon/awssdk/services/kinesis/SubscribeToShardUnmarshallingTest.java

```diff
@@ -81,37 +82,56 @@ public void setup() {
     }
 
     @Test
-    public void exceptionWithMessage_UnmarshalledCorrectly() throws InterruptedException {
+    public void exceptionWithMessage_UnmarshalledCorrectly() throws Throwable {
+        String errorCode = "ResourceNotFoundException";
         AbortableInputStream content = new MessageWriter()
             .writeInitialResponse(new byte[0])
-            .writeException("{\"message\": \"foo\"}")
+            .writeException("{\"message\": \"foo\"}", errorCode)
             .toInputStream();
 
         stubResponse(SdkHttpFullResponse.builder()
                                         .statusCode(200)
                                         .content(content)
+                                        .putHeader("x-amzn-requestid", REQUEST_ID)
                                         .build());
 
-        assertThatThrownBy(this::subscribeToShard)
-            .isInstanceOf(KinesisException.class)
-            .hasMessageContaining("foo");
+        try {
+            subscribeToShard();
+            fail("Expected ResourceNotFoundException exception");
+        } catch (ResourceNotFoundException e) {
+            assertThat(e.requestId()).isEqualTo(REQUEST_ID);
+            assertThat(e.statusCode()).isEqualTo(500);
+            assertThat(e.awsErrorDetails().errorCode()).isEqualTo(errorCode);
+            assertThat(e.awsErrorDetails().errorMessage()).isEqualTo("foo");
+            assertThat(e.awsErrorDetails().serviceName()).isEqualTo("kinesis");
```

Comment:
Where does this value come from? Is it unique?

Reply:
It comes from the service name defined in the client.
https://code.amazon.com/packages/AwsJavaSdk-Kinesis/blobs/a7781fca696c72c1ff9c81daa83e5ccc4d2b7c5e/--/src/software/amazon/awssdk/services/kinesis/KinesisAsyncClient.java#L88

Reply:
It comes from the service name generated in the client interface

Reply:
hmmm I don't think we should be using that value as it's not very unique and for some services it's vague like 'execute-api'. I'll create a issue in our backlog to address this as it's not related to this PR.

Reply:
I will remove the change to populate the service name then

Reply:
You can leave it in if you want. We are already exposing it for normal exceptions so we'll have to figure out the best way forward regardless.